### DISCUSSION TITLE
Render drawer beneath tabs stack

### DISCUSSION THREAD

**Discussion by whitestranger7** [react-navigation/react-navigation#9766] [Q&A]
Hello 👋

I wondering is there any solution for displaying drawer beneath tabs stack? The only solution I found using react-navigation is wrap every tab in `DrawerStack`, but the thing is that with such implementation every tab will include it's own drawer, which I particulary don't like at all. Regarding `TabsStack` flow we **need** to include only `<TabStack.screen  {...props} />` in it, because it would be rendered as tabs. So there's no way we could do something like this:

❌ - it wouldn't work
```js
const TabStack = createBottomTabNavigator()
const DrawerStack = createDrawerNavigator()

<TabStack.Navigator>
  <DrawerStack.Navigator>
     {TAB_ROUTES.map((tab) => (
      <DrawerStack.Screen
        {...tab}
      />
    ))}
  </DrawerStack.Navigator>
</TabStack.Navigator>
```

✅ - it would work, but creates 1 drawer per every tab
```js
<TabStack.Navigator>
   {TAB_ROUTES.map((tab, index) => (
    <TabStack.Screen
      {...tab}
      component={
      <DrawerStack.Navigator>
        <DrawerStack.Screen name={`DrawerItem-${index}`} component={tab.component}  />
      </Drawer.Stack.Navigator>
      }
    />
  ))}
</TabStack.Navigator>
```

So I wondering how I could implement something like that using 1 drawer.
![image](https://user-images.githubusercontent.com/36364993/126818439-d19dafaf-627d-462e-a25b-4bac279d5fd4.png)

**Comment by raajnadar** [comment]
Create a custom bottom bar and wrap it inside portal. Please try this, I think this will work.

**Comment by juanchaur** [comment]
@whitestranger7 did @raajnadar suggestion work?
